python/lancedb/context.py

```diff
@@ -137,17 +137,23 @@ def text_col(self, text_col: str) -> Contextualizer:
         self._text_col = text_col
         return self
 
-    def to_df(self) -> pd.DataFrame:
+    def to_df(self, threshold: int = 1) -> pd.DataFrame:
```

Comment:
I think this should be a parameter on the builder and not in the `to_df` method. This is mostly because we might offer a set of other `to_<something>` methods and we don't want to have to replicate those arguments over all of them.

Reply:
Hi Will, 
Thanks for pointing this out. I have updated the implementation accordingly.  Please let me know your thoughts over it.